chain/mempool_test.go

```diff
@@ -81,13 +128,76 @@ func TestCachedInputsAddInput(t *testing.T) {
 	require.True(ok)
 	require.Equal(tx.TxHash(), txid)
 
+	// Check the internal state. Since we've just added one input, there
+	// should be exactly one item in each map.
+	require.Len(c.inputs, 1)
+	require.Len(c.txids, 1)
+	require.Len(c.txids[tx.TxHash()], 1)
+
 	// Add the input again using the replacement tx.
 	c.addInput(op, replacedTx.TxHash())
 
 	// Lookup should now give us the replacement txid.
 	txid, ok = c.hasInput(op)
 	require.True(ok)
 	require.Equal(replacedTx.TxHash(), txid)
+
+	// Check the internal state.
+	//
+	// We should only have one input.
+	require.Len(c.inputs, 1)
+
+	// Expect two transactions.
+	require.Len(c.txids, 2)
+
+	// The new txid should be present.
+	require.Len(c.txids[replacedTx.TxHash()], 1)
+
+	// The old txid should be an empty map.
+	require.Empty(c.txids[tx.TxHash()])
```

Comment:
When would this old `txid` be removed? IIUC, we'd only want to notify/play the new one, since that is what would be available from `GetRawTransactions`.